tests/managed/methods.cs

```diff
@@ -88,6 +88,12 @@ public void Remove (Item item)
 		}
 	}
 
+	public class DuplicateMethods {
+		public int DoIt () { return 42; }
+		public int DoIt (int i) { return 42; }
+		public int DoIt (string s) { return 42; }	
```

Comment:
what happens with `public int DoIt (int i, int i) { return 84; }` ?
It's not a dupe of the signature so it should not fallback to types.

Another example would be
`bool Find (string name)` and
`bool Find (string firstName, lastName)`
It's clearly a win to use parameter names instead of types.